test/solidity/Facets/PolymerCCTPFacet.t.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import {Test} from "forge-std/Test.sol";
+import {PolymerCCTPFacet} from "lifi/Facets/PolymerCCTPFacet.sol";
+import {IPolymerCCTPFacet} from "lifi/Interfaces/IPolymerCCTP.sol";
+import {PolymerCCTPData} from "lifi/Interfaces/IPolymerCCTP.sol";
+import {ILiFi} from "lifi/Interfaces/ILiFi.sol";
+import {LibSwap} from "lifi/Libraries/LibSwap.sol";
+import {LibAllowList} from "lifi/Libraries/LibAllowList.sol";
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {InvalidReceiver, InvalidAmount} from "src/Errors/GenericErrors.sol";
+
+// Test version with allowlist management
+contract TestPolymerCCTPFacet is PolymerCCTPFacet {
+    constructor(address _tokenMessenger, address _usdc, address _polymerFeeReceiver)
+        PolymerCCTPFacet(_tokenMessenger, _usdc, _polymerFeeReceiver)
+    {}
+
+    function addDex(address _dex) external {
+        LibAllowList.addAllowedContract(_dex);
+    }
+
+    function setFunctionApprovalBySignature(bytes4 _signature) external {
+        LibAllowList.addAllowedSelector(_signature);
+    }
+}
+
+// Mock TokenMessenger
+contract MockTokenMessenger {
+    event DepositForBurn(uint256 amount, uint32 destinationDomain, bytes32 mintRecipient, address burnToken);
+
+    function depositForBurn(
+        uint256 amount,
+        uint32 destinationDomain,
+        bytes32 mintRecipient,
+        address burnToken,
+        bytes32,
+        uint256,
+        uint32
+    ) external {
+        // Transfer tokens from caller
+        ERC20(burnToken).transferFrom(msg.sender, address(this), amount);
+        emit DepositForBurn(amount, destinationDomain, mintRecipient, burnToken);
+    }
+}
+
+// Mock USDC
+contract MockUSDC is ERC20 {
+    constructor() ERC20("USD Coin", "USDC", 6) {}
+
+    function mint(address to, uint256 amount) external {
+        _mint(to, amount);
+    }
+}
+
+// Mock token for swap testing
+contract MockToken is ERC20 {
+    constructor(string memory name, string memory symbol, uint8 decimals) ERC20(name, symbol, decimals) {}
+
+    function mint(address to, uint256 amount) external {
+        _mint(to, amount);
+    }
+}
+
+// Mock DEX for swapping
+contract MockDEX {
+    function swap(address tokenIn, address tokenOut, uint256 amountIn, uint256 amountOut) external {
+        ERC20(tokenIn).transferFrom(msg.sender, address(this), amountIn);
+        ERC20(tokenOut).transfer(msg.sender, amountOut);
+    }
+}
+
+contract PolymerCCTPFacetTest is Test {
```

Comment:
We have TestBaseFacet.sol with standard tests for all bridge facets that could be leveraged here. 
We could also help you with this stuff once the contract is finalized.

Unit test line coverage must be 100%.